### DISCUSSION TITLE
redis密码无法正确解析

### DISCUSSION THREAD

**Discussion by liangkiller** [juicedata/juicefs#2844] [General]
**What happened**:
redis 5.6, 如果密码里有特殊字符，juicefs会报无法解析redis URL
**What you expected to happen**:
juicefs可以正确解析 redis 带有特殊字符的密码
**How to reproduce it (as minimally and precisely as possible)**:
redis 密码里有 特殊字符
**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:  juices 1.0
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

**Comment by zhijian-pro** [comment]
当 Redis 的用户名或者密码中包含特殊字符时需要将特殊字符通过 [URL encode](https://www.w3schools.com/tags/ref_urlencode.ASP) 的方式替换为 %xx 的格式，例如 @ 替换为 %40，或者使用环境变量的方式传递密码。
https://juicefs.com/docs/zh/community/databases_for_metadata#%E5%88%9B%E5%BB%BA%E6%96%87%E4%BB%B6%E7%B3%BB%E7%BB%9F
